src/main/java/net/glowstone/command/minecraft/TestForCommand.java

```diff
@@ -0,0 +1,88 @@
+package net.glowstone.command.minecraft;
+
+import net.glowstone.command.CommandTarget;
+import net.glowstone.command.CommandUtils;
+import net.glowstone.entity.GlowEntity;
+import net.glowstone.entity.GlowPlayer;
+import net.glowstone.io.entity.EntityStorage;
+import net.glowstone.util.mojangson.Mojangson;
+import net.glowstone.util.mojangson.ex.MojangsonParseException;
+import net.glowstone.util.nbt.CompoundTag;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.defaults.VanillaCommand;
+import org.bukkit.entity.Entity;
+
+import java.util.*;
+
+public class TestForCommand extends VanillaCommand {
+    public TestForCommand() {
+        super("testfor",
+                "Tests for a certain target in game",
+                "/testfor <player> [dataTag]",
+                Collections.emptyList());
+        setPermission("minecraft.command.testfor");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String label, String[] args) {
+        if (!testPermission(sender)) {
+            return false;
+        }
+
+        if (args.length < 1) {
+            sender.sendMessage(ChatColor.RED + "Usage:" + usageMessage);
+            return false;
+        }
+
+        String name = args[0];
+        Entity[] entities;
+        if (name.startsWith("@")) {
+            CommandTarget target = new CommandTarget(sender, name);
+            entities = target.getMatched(CommandUtils.getLocation(sender));
```

Comment:
Assuming the `/testfor` command can only be executed by physical senders (no console), you should validate that before (see other commands that use target selectors)

Reply:
I did some testing with the vanilla Minecraft server, and /testfor works fine from the console. Is there anything different I have to do in that case?

Reply:
If it can be executed by the console, I assume you can't use target selectors (like `@e`) as the console? IIRC you can only test-for players using the console.

Reply:
After more testing, `@e` does work in the console. Just spawned a bunch of zombies into my vanilla server, and ran `/testfor @e[r=500,type=zombie]`, and it gave me *[Server: Found Zombie]*. Looks like the command uses (0, 0, 0) as the location for the command when run in the console.

Reply:
That's interesting, I assumed it would take the target string as a literal player name if it's a console. 

Did you test this with Vanilla or Spigot? Since Vanilla can only have 1 world, it makes sense to have this arbitrary location. But since Spigot (and Glowstone) support multiple worlds, this may not be possible (unless it takes the default world too?)

Reply:
I was testing with the Vanilla server. Just spun up a Spigot server, spawned in some zombies, and tested the command, works there too. Again, its centered on (0, 0, 0) in the default overworld.

Reply:
Okay. So that means we should return `0 0 0` here:
https://github.com/GlowstoneMC/Glowstone/blob/47db4c461f2077620731692777c07e146af4d3f1/src/main/java/net/glowstone/command/CommandUtils.java#L136

and the default world here:

https://github.com/GlowstoneMC/Glowstone/blob/47db4c461f2077620731692777c07e146af4d3f1/src/main/java/net/glowstone/command/CommandUtils.java#L127

Reply:
Will do. Want me to clean up the other commands that use these functions and check for null, or do you want that to be in a separate PR?

Reply:
That's up to you, and what scope you want to set for your PR :)